examples/bot/main.rs

```diff
@@ -18,24 +18,30 @@ impl rlbot::Bot for MyBot {
         self.player_index = index;
     }
 
-    fn tick(&mut self, packet: &rlbot::LiveDataPacket) -> rlbot::PlayerInput {
+    fn tick(&mut self, packet: &rlbot::GameTickPacket) -> rlbot::ControllerStateArgs {
         get_input(self.player_index, packet)
     }
 }
 
-fn get_input(player_index: usize, packet: &rlbot::LiveDataPacket) -> rlbot::PlayerInput {
-    let ball = packet.GameBall;
-    let ball_loc = Vector2::new(ball.Physics.Location.X, ball.Physics.Location.Y);
-    let car = packet.GameCars[player_index];
-    let car_loc = Vector2::new(car.Physics.Location.X, car.Physics.Location.Y);
+fn get_input(player_index: usize, packet: &rlbot::GameTickPacket) -> rlbot::ControllerStateArgs {
+    let ball = packet.ball().expect("Missing ball");
```

Comment:
Yeah I tried that and yes found out it's not something that's possible yet.

The other way this could have been solved honestly is by having structs instead of tables in the flatbuffer schema. Eg the top level could be tables to allow some evolution, but structs inside for all the data that is unlikely to ever change in structure `ball.physics.location for example`.

Then we wouldn't need all the `expect`. But it's probably too late for that as there is so much code out there already using the existing schema.

Another way we could reduce verbosity is by deserializing into a regular struct in the end. But I don't think the rlbot-rust lib should do this, seems presumptuous about what users will want.

Reply:
I think it's reasonable to deserialize to a better data structure as part of the high-level interface. The same way that if we were making a library for some webservice like Twilio, we wouldn't return a `serde_json::Value` everywhere, we would deserialize to `twilio::PhoneNumber` (or w/e). After all flatbuffer is just a serialization format like JSON is.

Reply:
Right I'm mostly just wary of having 3 different ways to represent the data, ffi, flat and this new one. But I think we can further motivate this further by making these structs more "rustic", ie following rust conventions much more closely. So I can get behind that. Separate PR of course!